odl/discr/grid.py

```diff
@@ -802,346 +947,104 @@ def __array__(self, dtype=None):
 
     def __repr__(self):
         """Return ``repr(self)``."""
-        inner_str = ', '.join(array1d_repr(vec) for vec in self.coord_vectors)
-        return '{}({})'.format(self.__class__.__name__, inner_str)
+        if self.is_uniform:
+            constructor = 'uniform_grid'
+            posargs = [list(self.min_pt), list(self.max_pt), self.shape]
+            inner_str = signature_string(posargs, [])
+            return '{}({})'.format(constructor, inner_str)
+        else:
+            constructor = self.__class__.__name__
+            posargs = [array1d_repr(v) for v in self.coord_vectors]
+            inner_str = signature_string(posargs, [], sep=[',\n', ', ', ', '],
+                                         mod=['!s', ''])
+            return '{}(\n{}\n)'.format(constructor, indent_rows(inner_str))
 
     def __str__(self):
         """Return ``str(self)``."""
-        grid_str = ' x '.join(array1d_str(vec) for vec in self.coord_vectors)
+        posargs = [array1d_repr(v) for v in self.coord_vectors]
+        grid_str = signature_string(posargs, [], sep=[' x ', ', ', ', '],
+                                    mod=['!s', ''])
         return 'grid ' + grid_str
 
 
-class RegularGrid(TensorGrid):
```

Comment:
![74898985](https://cloud.githubusercontent.com/assets/5030250/22169237/e0fd6680-df72-11e6-9279-f6d69e2698cc.jpg)

Meh, not really..